docs/src/0-quickstart.jl

```diff
@@ -0,0 +1,274 @@
+
+# Copyright (c) 2023, University of Luxembourg                             #src
+# Copyright (c) 2023, Heinrich-Heine University Duesseldorf                #src
+#                                                                          #src
+# Licensed under the Apache License, Version 2.0 (the "License");          #src
+# you may not use this file except in compliance with the License.         #src
+# You may obtain a copy of the License at                                  #src
+#                                                                          #src
+#     http://www.apache.org/licenses/LICENSE-2.0                           #src
+#                                                                          #src
+# Unless required by applicable law or agreed to in writing, software      #src
+# distributed under the License is distributed on an "AS IS" BASIS,        #src
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. #src
+# See the License for the specific language governing permissions and      #src
+# limitations under the License.                                           #src
+
+# # Quick start
+#
+# The primary purpose of ConstraintTrees.jl is to make the representation of
+# constraint systems neat, and thus make their manipulation easy and
+# high-level. In short, the package abstracts the users from keeping track of
+# variable and constraint indexes in matrix form, and gives a nice data
+# structure that describes the system, while keeping all variable allocation
+# and constraint organization completely implicit.
+#
+# Here we demonstrate the absolutely basic concepts on the "field allocation"
+# problem.
+#
+# ## The problem: Field area allocation
+#
+# Suppose we have 100 square kilometers of field, 500 kilos of fertilizer and
+# 300 kilos of insecticide. We also have a practically infinite supply of wheat
+# and barley seeds. If we decide to sow barley, we can make 550🪙 per square
+# kilometer of harvest; if we decide to go with wheat instead, we can make
+# 350🪙. Unfortunately each square kilometer of wheat requires 6 kilos of
+# fertilizer, and 1 kilo of insecticide, whereas each square kilometer of
+# barley requires 2 kilos of fertilizer and 4 kilos of insecticide, because
+# insects love barley.
+#
+# How much of our fields should we allocate to wheat and barley to maximize
+# our profit?
+#
+# ## Field area allocation with ConstraintTrees.jl
+#
+# Let's import the package and start constructing the problem:
+
+import ConstraintTrees as C
+
+# Let's name our system `s`. We first need a few variables:
+
+s = C.variables(keys = [:wheat, :barley])
+
+# With ConstraintTrees.jl, we can (and want to!) label everything very nicely
+# -- the constraint trees are essentially directory structures, so one can
+# prefix everything with symbols to put it into nice directories.
+
+s = :area^s
```

Comment:
this might confuse some, why is `s` being overwritten here?

Reply:
true